### DISCUSSION TITLE
Adafruit Font invisible within menu after upgrading to tcMenu Designer 3.0.0

### DISCUSSION THREAD

**Discussion by mmilesg** [davetcc/tcMenu#281] [Q&A] [answered]
Hi Dave,
   I just upgraded my designer app and requisite libraries, and it seems that my menu is now unable to display the Adafruit font while in the menu (after regenerating the code from the UI). I just see blank lines with the little cursor/edit icon indicating the highlighted item. the same font is used when I takeOverDisplay() and shows perfectly in my UI functions outside of the menu. No use of setFont() or anything that might change the defaults are in place.

I tried different default fonts from Adafruit thinking that some of my character customizations might be incompatible, but no change.

I also tried ticking the "Use tcUnicode..." option just to see if any fonts would work and that DID work, but of course the font sizing/spacing was all wrong for my UI.

I also tried deleting all 5 of my generated code files so the UI would create fresh copies of everything. No change.

My thought was that maybe my emf was bad or somehow incompatible with the new UI version since it was created with an earlier one (I've attached it so you can take a look) [SR_Controller_SureServo2.zip](https://github.com/davetcc/tcMenu/files/10406931/SR_Controller_SureServo2.zip)

Otherwise, is there any further info you need or some new way I need to start the menu with these versions that I may have missed?

These are the library versions I have installed:
adafruit/Adafruit BusIO@^1.14.1
adafruit/Adafruit GFX Library@^1.11.3
adafruit/Adafruit ILI9341@^1.5.12
adafruit/Adafruit TouchScreen@^1.1.3
davetcc/IoAbstraction@^3.0.2
davetcc/LiquidCrystalIO@^1.4.3
davetcc/SimpleCollections@^1.2.3
davetcc/TaskManagerIO@^1.3.5
davetcc/TcMenu@^3.0.1
davetcc/tcUnicodeHelper@^0.2.0

Thank you very much for any insight you can provide.

**Comment by davetcc** [comment]
Thanks, I'll try your project over the weekend and feed back. In the mean time please continue to use 2.5.0 if that doesn't have the issue. Ensure that you also set tcMenuLib to 2.5.0 and IoAbstraction to 2.4.3.

**Reply by mmilesg** [reply]
Thanks Dave,
   I'll hold off on the downgrade so I can help provide more info and test things for you, if you have any suggestions after taking a look.
Here's an image which shows the symptoms better:
![image](https://user-images.githubusercontent.com/94007807/212389691-479ea317-25f8-4590-a187-a4ee6aebdc29.png)

Also, it occurs to me that I may have inflicted this glitch upon myself; I have manually edited the EMF file to keep all my menu item IDs in ascending order. I don't believe I made any mistakes and everything appears to be correct when opening the file from the UI, but I guess it's something to keep in mind. Perhaps a "renumber all menu items" feature could be useful in keeping all your more pedantic users happy (talking exclusively about myself here.) 

Take your time. I'm happy to work on other areas of my program in the mean time. Your help is always appreciated whenever I can get it.

Thanks!

**Comment by mmilesg** [comment]
Follow up:
 
I ended up needing to change some things in my menu so I took the opportunity to do some further testing.

Just to make sure I didn't have any issues with the project file that was originally created before Designer 3.xxx I went ahead and started a new project from scratch (creating each new item from within Designer 3.0.0 while referencing the old file) Unfortunately, I still cannot see any of the menu items with the new project file (exactly as shown in the image above.)

I also went so far as to temporarily comment out all commands like  setVisible(); notifyStructureChanged()  and takeOverDisplay() so the menu just displays directly after setupMenu();.  Still no menu text visible.

Another quirk: I don't know if this is intentional, or perhaps related to the issues I'm having, but I did notice that the "Set Font" dialog screen is slightly different depending on which file I'm working with (editing both old and new files from within Designer 3.0.0)

![image](https://user-images.githubusercontent.com/94007807/213818606-7bb6c8a0-36df-4bdf-9250-c60c9254b1ca.png)
Old file shown on left allows to select font format, new file on right allows font size.

Hope this helps! Thanks again.

**Comment by davetcc** [comment]
Agreed, it is broken :(

Looking at why now, hopefully, a trivial fix.

**Comment by davetcc** [comment, marked as answer]
See https://github.com/davetcc/tcMenu/issues/276 this fixes the issue for now, it will be released shortly in 3.1

**Reply by mmilesg** [reply]
Awesome! I didn't realize that these two issues were related, but I had some suspicions as I was looking over everything. So glad you could pinpoint it and kill two birds with one stone. Thanks so much!

**Reply by mmilesg** [reply]
Followup: Yep! just replaced that line in tcMenuAdaFruitGfx.cpp and everything is apparently working as normal.

**Comment by davetcc** [comment]
Thanks for reporting this
